From stern@rowland.harvard.edu Thu Apr  5 13:07:54 2007
From: Alan Stern <stern@rowland.harvard.edu>
Date: Thu, 5 Apr 2007 16:07:44 -0400 (EDT)
Subject: USB: remove unneeded WARN_ON
To: Greg KH <greg@kroah.com>, David Brownell <david-b@pacbell.net>
Cc: USB development list <linux-usb-devel@lists.sourceforge.net>
Message-ID: <Pine.LNX.4.44L0.0704051606540.2495-100000@iolanthe.rowland.org>


This patch (as883) removes an out-of-date WARN_ON from the main HCD
endpoint-disable routine.  The warning is triggered whenever an
endpoint is disabled while the root hub is suspended.  In the past
that may not have been legal, but it definitely is legal now.  Merely
unbinding a USB driver will do it.

Furthermore, I've never seen any occurrences of this warning that
really did signal an actual bug or error condition.  At this point it
has outlived its purpose.

Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>

---
 drivers/usb/core/hcd.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/usb/core/hcd.c
+++ b/drivers/usb/core/hcd.c
@@ -1175,10 +1175,6 @@ void usb_hcd_endpoint_disable (struct us
 	struct urb		*urb;
 
 	hcd = bus_to_hcd(udev->bus);
-
-	WARN_ON (!HC_IS_RUNNING (hcd->state) && hcd->state != HC_STATE_HALT &&
-			udev->state != USB_STATE_NOTATTACHED);
-
 	local_irq_disable ();
 
 	/* ep is already gone from udev->ep_{in,out}[]; no more submits */
